packages/service/lib/permission.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  type Class,
+  Classes,
+  type Course,
+  type CourseId,
+  type Role,
+  type User,
+} from "../models";
+import { ClassPermissionError, CoursePermissionError } from "./error";
+
+export function assertInCourse(
+  user: User,
+  course: Course | CourseId,
```

Comment:
```suggestion
  course: CourseId,
```

`CourseId` is just a subtype of `Course`.